libcodechecker/generic_package_context.py

```diff
@@ -23,6 +24,36 @@
 LOG = logger.get_logger('system')
 
 
+class SeverityMap(Mapping):
+    """
+    A dictionary which maps checker names to severity levels.
+    If a key is not found in the map and the checker name is a compiler warning
+    it will return severity level of MEDIUM.
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.store = dict(*args, **kwargs)
+
+    def __getitem__(self, key):
+        # Key is not specified in the store and it is a compiler warning.
+        if key not in self.store and key.startswith('clang-diagnostic-'):
+            return "MEDIUM"
+
+        return self.store[key]
```

Comment:
Is it possible that the `key` is not in `self.store` and it doesn't start with `clang-diagnostic-`? Because in this case a `KeyError` is raised.